### PR TITLE
Slack on-boarding fix

### LINKED ISSUE #46: Slack users can onboard multiple times
Slack users can currently onboard multiple times. Checks need to be added that ensure that the same slack id cannot be used for multiple users.

### PR DESCRIPTION
**Pull request by kbristow**
Added duplication checks to prevent slack users on-boarding multiple times.

Resolves #46

### CODE REVIEW

File: nucleus/src/main/java/za/co/absa/subatomic/application/member/TeamMemberService.java [hunk 1]
```diff
@@ -57,9 +57,28 @@ public String newTeamMember(String firstName, String lastName,
...
+            String email, String domainUsername, String screenName,
             String userId) {
+        TeamMemberEntity existingMember = this.findByEmail(email);
+        if (existingMember != null) {
+            throw new DuplicateRequestException(MessageFormat.format(
+                    "Requested email address {0} is already in use.",
+                    email));
+        }
 
+        existingMember = this.findByEmail(email);
```
Comment (KieranHons) [inline nucleus/src/main/java/za/co/absa/subatomic/application/member/TeamMemberService.java]: Should this not be findByDomainUsername?

Comment (kbristow) [reply nucleus/src/main/java/za/co/absa/subatomic/application/member/TeamMemberService.java]: It should be

Comment (kbristow) [reply nucleus/src/main/java/za/co/absa/subatomic/application/member/TeamMemberService.java]: Fixed

Comment (kbristow) [reply nucleus/src/main/java/za/co/absa/subatomic/application/member/TeamMemberService.java]: Yes. Fixed now.

**Pull request merged**